routes/course.js

```diff
@@ -205,7 +206,28 @@ router.patch('/api/courses/dragged', auth, async (req, res) => {
       }).exec();
       // check if course is held at new term!
       if (!checkDestValid(sisCourses, course, newTerm)) {
-        return errorHandler(res, 400, { message: 'no course this semester' });
+        return errorHandler(res, 400, {
+          message: "Course isn't usually held this semester!",
+        });
+      }
+      // check if course is a duplicate at new term!
+      const retrievedCourses = await Courses.findByPlanId(course.plan_id);
```

Comment:
alternative: query Courses for plan id, year, title, and number. that way you can remove the for loop below. 

idk what is better :?

Reply:
That makes a lot of sense! I think your approach would be faster than a for loop